code/modules/organs/external/_external.dm

```diff
@@ -95,9 +95,12 @@
 		for(var/obj/item/organ/O in internal_organs)
 			qdel(O)
 
+	if(owner)
+		owner.organs -= src
+		owner.organs_by_name -= src.name
```

Comment:
Not `src.name`, but `src.organ_tag`
Organs with name "left leg" and one with "robotic left leg" has same `organ_tag` (L_LEG)